curator-framework/src/test/java/org/apache/curator/framework/state/TestConnectionStateManager.java

```diff
@@ -92,6 +94,7 @@ public void stateChanged(CuratorFramework client, ConnectionState newState)
 
     @Test
     public void testConnectionStateRecoversFromUnexpectedExpiredConnection() throws Exception {
+        assumeTrue(() -> (getMajor() == 3 && getMinor() >= 6) || (getMajor() > 4), "Zookeeper version must be 3.6 or higher");
```

Comment:
This does not really work...
It seems javac inlines the values of the constants and the test still prints `3` and `6` even for `curator-test-zk35` modole.